### ISSUE TITLE
[Bug?] unknown encoding ASCII-8BIT

### ISSUE THREAD

**Issue by bogdan** [sparklemotion/nokogiri#553]
I am using ruby 1.9.3-preview1 and nokogiri 1.5.0.

``` ruby
require "nokogiri"

s = ""
puts Nokogiri::HTML::DocumentFragment.parse("e#{s}").to_s.inspect
```

Outputs:

```
output error : unknown encoding ASCII-8BIT
""
```

It is very strange that if empty string is inserted inline `"e#{""}"` - there is no error.

libxml `2.7.8.dfsg-2ubuntu0.1`

**Comment by flavorjones** [comment]
Closing, never could reproduce. Happy to re-open if someone can help me reproduce the problem.

**Comment by amatsuda** [comment]
@flavorjones Reproducing the problem is not hard.

```ruby
require 'nokogiri'

s = String.new 'hello', encoding: Encoding::ASCII_8BIT

p xml: Nokogiri::XML::DocumentFragment.parse(s).to_s
#=> {:xml=>"hello"}
p html: Nokogiri::HTML::DocumentFragment.parse(s).to_s
#=> output error : unknown encoding ASCII-8BIT
#=> {:html=>""}
```

Seems like `native_write_to` does not accept `'ASCII-8BIT'`?

**Comment by flavorjones** [comment]
@amatsuda Thank you for helping me reproduce.

**Comment by stephankaag** [comment]
Any update available?

**Comment by pmackay** [comment]
I'm getting this only on Heroku on ruby 2.3.3. Is there any way to workaround it? 

@yagudaev did you find any more specifics about why it would just happen on Heroku?

**Comment by flavorjones** [comment]
@pmackay this is reproducible in ruby 2.4.0. It's definitely not heroku-specific or 2.3.3-specific.

**Comment by flavorjones** [comment]
It's also specific to HTML::DocumentFragment. A potential workaround for now is to use HTML::Document or else XML::DocumentFragment if possible.

**Comment by envygeeks** [comment]
Since `HTML::Document.parse` deems it necessary to add a document type, and that document type is older than most modern browsers, it's not really a work around any of those.  We have no way to truly parse a real fragment without arbitrary warnings, or DOCTYPE.

Is this happening downstream or within Nokogiri?

**Comment by flavorjones** [comment]
This is remarkably similar to #1659 which was fixed by #1674. I imagine there's a similar problem here, if anyone wants to help out and take a look.
